src/main/test/surfrdUtils_test/test_surfrdUtils.pf

```diff
@@ -45,74 +45,105 @@ contains
 
   @Test
   subroutine test_collapse_nat_pfts()
-     ! This test tries 3 values of n_dom_pfts:
+     ! This test tries 5 values of n_dom_pfts:
      ! 1) 0, which ends up with the same output values as were used as inputs
-     ! 2) 2, which collapses the vegetation to 2 dominant pfts
-     ! 3) natpft_size, which behaves the same as (1) whether subroutine
+     ! 2) 1, which collapses the vegetation to 1 dominant pft
+     ! 3) 2, which collapses the vegetation to 2 dominant pfts
+     ! 4) 3, which collapses the vegetation to 3 dominant pfts
+     ! 5) natpft_size, which behaves the same as (1) whether subroutine
      ! collapse_nat_pfts actively collapses the pfts to themselves or simply
      ! skips the work with an if statement (current implementation).
      !
-     ! For each value of n_dom_pfts, we test two pft configurations that
-     ! need to be collapsed. One in which the maximum weight is in pft5 and one
-     ! in which the maximum weight is in pft0.
+     ! For each value of n_dom_pfts, we test three pft configurations that
+     ! need to be collapsed.
      use pftconMod, only: pftcon
      use clm_instur, only: wt_nat_patch
      use clm_varpar, only: natpft_ub
 
      implicit none
      integer :: test  ! test number in loop of tests
-     integer, parameter :: begg = 2, endg = 3, natpft_size = 15, tests = 3
-     integer, parameter :: n_dom_pfts(tests) = (/ 0, 2, natpft_size /)
+     integer, parameter :: begg = 2, endg = 4, natpft_size = 15, tests = 5
+     integer, parameter :: n_dom_pfts(tests) = (/ 0, 1, 2, 3, natpft_size /)
      real(r8), allocatable :: wt_nat_patch_expected(:,:)
-     real(r8) :: expctd_1, expctd_2
+     real(r8), allocatable :: wt_nat_patch_in_out(:,:)  ! used in subr. call
+     real(r8) :: expctd(9)
 
      ! Set relevant pftcon values to defaults; override where necessary
      call pftcon%InitForTesting()
      natpft_ub = natpft_size - 1
      allocate( wt_nat_patch(begg:endg,0:natpft_ub) )
      allocate( wt_nat_patch_expected(begg:endg,0:natpft_ub) )
+     allocate( wt_nat_patch_in_out(begg:endg,0:natpft_ub) )
```

Comment:
I caught a problem while updating this subroutine:
Without this new variable wt_nat_patch_in_out, my wt_nat_patch got overwritten in the subroutine call below.

Introducing additional n_dom_pfts values to be tested caused the unit test to fail leading me to this problem.